opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -8,19 +10,21 @@ use std::{
 
 use crate::metrics::data::{self, Aggregation, DataPoint, Temporality};
 use crate::metrics::AttributeSet;
+use once_cell::sync::Lazy;
 use opentelemetry::KeyValue;
 use opentelemetry::{global, metrics::MetricsError};
 
-use super::{
-    aggregate::{is_under_cardinality_limit, STREAM_OVERFLOW_ATTRIBUTE_SET},
-    AtomicTracker, Number,
-};
+use super::{aggregate::is_under_cardinality_limit, AtomicTracker, Number};
+
+pub(crate) static STREAM_OVERFLOW_ATTRIBUTES: Lazy<Vec<KeyValue>> =
+    Lazy::new(|| vec![KeyValue::new("otel.metric.overflow", "true")]);
 
 /// The storage for sums.
 struct ValueMap<T: Number<T>> {
-    values: RwLock<HashMap<AttributeSet, T::AtomicTracker>>,
+    values: RwLock<HashMap<Vec<KeyValue>, Arc<T::AtomicTracker>>>,
     has_no_value_attribute_value: AtomicBool,
     no_attribute_value: T::AtomicTracker,
+    count: AtomicUsize,
```

Comment:
count is tracked separately, as number of entries in HashMap is no longer accurate, as we insert original+sorted in hashmap.